dask_expr/_expr.py

```diff
@@ -699,8 +702,10 @@ def dtypes(self):
     def _meta(self):
         raise NotImplementedError()
 
-    def __dask_graph__(self):
+    def _materialize(self):
         """Traverse expression tree, collect layers"""
+        from distributed.scheduler import ensure_materialization_allowed
+        ensure_materialization_allowed()
```

Comment:
This is mostly for ad-hoc testing. If we find it valuable we can actually add this as a proper API

Reply:
This would add distributed as a hard dependency, wouldn't it? That's something I'd rather avoid

Reply:
🔝 Draft ;)

Reply:
**If** we implemented an API like this, this would need to live in dask/dask